### ISSUE TITLE
Enhancement request - Chinese support

### ISSUE THREAD

**Issue by honken** [mvysny/aedict#806]
The app is fantastic - clearly lovingly made with tons of features.  Probably my no. 1 feature is the export to AnkiDroid deck option - because as much as I love AnkiDroid, making your own cards is a fairly horrible process, especially on mobile.

Which brings me to the request - Chinese support.  As someone who is fairly fluent with Japanese and looking to pick up Chinese, I have yet to find any dictionary app that compares to Aedict.  I have no idea how much work this would be to integrate but there is already a well-maintained Chinese EDICT dictionary on a Creative Commons license (below) which perhaps could be used for basic functionality. In 2017 where Chinese is more relevant than ever, I think there is a significant audience for this.

Thanks for considering!

https://www.mdbg.net/chinese/dictionary?page=cedict

**Comment by mvysny** [comment]
Hi, thanks for the praise, I'm happy that you like the app :)

Now for the Chinese support. I've looked at the CEDICT file and it seems quite straightforward and easy to parse. I can thus convert it to JMDict format easily, and then parse by Aedict. I'd need to relax some checks first (e.g. reading must be in kana) but that's all doable.

The problem lies somewhere else. Aedict contains a lot of Japanese grammar hardwired - it performs sentence analysis with verb deinflection, shows verb inflection charts, categorizes verbs based on Japanese verb categories etc. Passing in Chinese verbs into those algorithms will produce weird output at best, or it will outright fail at worst.

To add the support for Chinese at the level of the support of Japanese, I would have to add Chinese grammar-specific things, which is impossible for me since I do not know the Chinese grammar at all. Also, I expect that there are already lots of awesome Chinese dictionaries; even if I created a new one it would be lacking behind feature-wise.

So, from this perspective it doesn't make much sense to add Chinese dictionary to Aedict, but please let me know your thoughts on this.

**Comment by honken** [comment]
Hi and thank you for responding so quickly!

You make a great point, to introduce Chinese and bring it up to the level of the existing Japanese support would be a significant undertaking, they are grammatically very different languages as far as I can tell (from my limited Chinese study so far).

Having said that, just having access to just a portion of Aedict's features like basic offline lookup, Notepad and AnkiDroid export would be phenomenally useful to me and I'm sure others. It could even be behind a "beta feature" checkbox in options if you're concerned about it "tarnishing the brand", so to speak.

To your other point, you really would think there would be some great Chinese dictionaries out there... But the truth is I am very much struggling to find one for Android that isn't either extremely basic, online only and/or littered with ads, and not a single one with the "holy grail" of AnkiDroid support.

So in summary I would say, "do not let perfect be the enemy of good" - even if it's not feature matched with Japanese (they are different languages after all), it would *still* be better than anything else out there as far as I'm concerned.  Many thanks again for taking the time to review.

PS - I could have sworn there used to be an option buried away somewhere regarding pinyin display (e.g. yín vs. yin2), or am I getting mixed up? Cheers,

**Comment by mvysny** [comment]
Yeah, every kanji also has pinyin reading; the setting is in Settings / Romanization,Kanjipad / Pinyin Romanization.

To be honest I'm not afraid of the "brand tarnish" :-) I'm afraid about having pinyin as reading since all of the Aedict code expects hiragana or katakana there. I'm afraid about somebody accidentally activating the dictionary and then reporting crashes all over the place. But, perhaps if there was a big warning on this dictionary then it could be fun to test it out ;)

Let me try to create a prototype.

**Comment by mvysny** [comment]
The CEDICT has been indexed, uploaded to the dictionary server and is available for download. The support is EXPERIMENTAL and often illogical findings will be reported, for example JLPT categorization; also the frequency (common/uncommon) is absolutely off and should be completely ignored. If you find other issues, please open new bug reports for those. If you like the feature, please consider supporting me at https://www.patreon.com/aedict . Thanks and enjoy :-)
